[Fix] point cloud export broken in Dockerfile

Zunhammer opened the pull request:
Rebased the docker image to one built on ubuntu 22.04 where the libraries are matching the required versions for pymeshlab. Pointcloud export is now working again.

The new image comes with some additional changes (only a side node):
Python is now version 3.10
CUDA is now version 11.7
Colmap needs to be compiled with c++14

Didn't find any issues with the new versions so far.

Fixes #1129

tancik commented on Dockerfile:
```diff
@@ -1,5 +1,5 @@
 # Define base image.
-FROM nvidia/cudagl:11.3.1-devel
+FROM nvidia/cuda:11.7.1-devel-ubuntu22.04
```
Will this still work with windows?

Zunhammer replied:
Yes, also the old image was based on Ubuntu (version 20.04), you simply need the docker environment based on WSL and Nvidia drivers as before.
Actually, this shouldn't make any difference from the user perspective.

tancik replied:
Gotcha, (docker beginner here)

tancik reviewed (commented):
Thanks for looking into this. The documentation should also be updated to reflect the new prerequisites. -nvidia/cuda:11.7.1-devel-ubuntu22.04

gowriaddepalli commented:
When will this merge in the main branch?

gowriaddepalli commented:
How to get version of python 3.10 + cuda 11.6 to run on google colab demo for nerfstudio? Any idea? I can change cuda 11.3 to 11.6 on google colab. Please let me know. Thanks in advance. It can help me fix ns-export issue maybe @tancik

Zunhammer commented:
> How to get version of python 3.10 + cuda 11.6 to run on google colab demo for nerfstudio? Any idea? I can change cuda 11.3 to 11.6 on google colab. Please let me know. Thanks in advance. It can help me fix ns-export issue maybe @tancik

Sorry, I have no experience with Colab, also I don't see this related to the Dockerfile/docker image.

Zunhammer commented:
> Thanks for looking into this. The documentation should also be updated to reflect the new prerequisites.

True, I updated the description and added some more hints there. The only requirement that chaned is actually the nvidia driver which needs to be compatible with CUDA 11.7.

tancik reviewed (approved):
LGTM
